1.11.1: build fails with `[-Werror=odr]` when LTO is used

### Describe the bug
Looks like LTO exposes some issues.
Build fails with `[-Werror=odr]` when LTO is used.
I'm using gcc 11.2.1.
```console
make[3]: Leaving directory '/home/tkloczko/rpmbuild/BUILD/ucx-1.11.1/test/gtest/ucm/test_dlopen'
make[3]: Entering directory '/home/tkloczko/rpmbuild/BUILD/ucx-1.11.1/test/gtest'
/bin/sh ../../libtool  --tag=CXX   --mode=link /usr/bin/g++ -O3 -g -Wall -Werror -funwind-tables -Wno-missing-field-initializers -Wno-unused-parameter -Wno-unused-label -Wno-long-long -Wno-endif-labels -Wno-sign-compare -Wno-multichar -Wno-deprecated-declarations -Winvalid-pch -fno-tree-vectorize -DGTEST_UCM_HOOK_LIB_DIR="\"/home/tkloczko/rpmbuild/BUILD/ucx-1.11.1/test/gtest/ucm/test_dlopen/.libs\""  -O2 -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fdata-sections -ffunction-sections -flto=auto -flto-partition=none  -no-install -Wl,-dynamic-list-data  -Wl,-z,relro -Wl,--as-needed -Wl,--gc-sections -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -flto=auto -flto-partition=none -fuse-linker-plugin -o gtest common/gtest-gtest-all.o common/gtest-main.o common/gtest-test_gtest_cmn.o common/gtest-mem_buffer.o common/gtest-test_helpers.o common/gtest-test_obj_size.o common/gtest-test_watchdog.o common/gtest-test_perf.o common/gtest-test.o ucm/gtest-malloc_hook.o uct/gtest-test_amo.o uct/gtest-test_amo_add_xor.o uct/gtest-test_amo_and_or.o uct/gtest-test_amo_cswap.o uct/gtest-test_amo_fadd_fxor.o uct/gtest-test_amo_fand_for.o uct/gtest-test_amo_swap.o uct/gtest-test_event.o uct/gtest-test_fence.o uct/gtest-test_flush.o uct/gtest-test_many2one_am.o uct/gtest-test_md.o uct/gtest-test_mm.o uct/gtest-test_mem.o uct/gtest-test_p2p_am.o uct/gtest-test_p2p_err.o uct/gtest-test_p2p_mix.o uct/gtest-test_p2p_rma.o uct/gtest-test_pending.o uct/gtest-test_progress.o uct/gtest-test_uct_ep.o uct/gtest-test_uct_perf.o uct/v2/gtest-test_uct_query.o uct/gtest-test_zcopy_comp.o uct/gtest-uct_p2p_test.o uct/gtest-uct_test.o uct/gtest-test_stats.o ucs/gtest-test_event_set.o ucs/gtest-test_stats_filter.o uct/gtest-test_peer_failure.o uct/gtest-test_tag.o uct/tcp/gtest-test_tcp.o ucp/gtest-test_ucp_am.o ucp/gtest-test_ucp_ep.o ucp/gtest-test_ucp_stream.o ucp/gtest-test_ucp_peer_failure.o ucp/gtest-test_ucp_atomic.o ucp/gtest-test_ucp_dt.o ucp/gtest-test_ucp_tls.o ucp/gtest-test_ucp_memheap.o ucp/gtest-test_ucp_mmap.o ucp/gtest-test_ucp_mem_type.o ucp/gtest-test_ucp_perf.o ucp/gtest-test_ucp_proto.o ucp/gtest-test_ucp_rma.o ucp/gtest-test_ucp_rma_mt.o ucp/gtest-test_ucp_tag_cancel.o ucp/gtest-test_ucp_tag_match.o ucp/gtest-test_ucp_tag_offload.o ucp/gtest-test_ucp_tag_mt.o ucp/gtest-test_ucp_tag_perf.o ucp/gtest-test_ucp_tag_probe.o ucp/gtest-test_ucp_tag_xfer.o ucp/gtest-test_ucp_tag_mem_type.o ucp/gtest-test_ucp_tag.o ucp/gtest-test_ucp_context.o ucp/gtest-test_ucp_worker.o ucp/gtest-test_ucp_wireup.o ucp/gtest-test_ucp_request.o ucp/gtest-test_ucp_wakeup.o ucp/gtest-test_ucp_fence.o ucp/gtest-test_ucp_sockaddr.o ucp/gtest-ucp_test.o ucp/gtest-ucp_datatype.o ucs/gtest-test_algorithm.o ucs/gtest-test_arbiter.o ucs/gtest-test_async.o ucs/gtest-test_bitmap.o ucs/gtest-test_callbackq.o ucs/gtest-test_class.o ucs/gtest-test_config.o ucs/gtest-test_conn_match.o ucs/gtest-test_datatype.o ucs/gtest-test_bitops.o ucs/gtest-test_debug.o ucs/gtest-test_memtrack.o ucs/gtest-test_math.o ucs/gtest-test_mpmc.o ucs/gtest-test_mpool.o ucs/gtest-test_pgtable.o ucs/gtest-test_profile.o ucs/gtest-test_rcache.o ucs/gtest-test_memtype_cache.o ucs/gtest-test_stats.o ucs/gtest-test_strided_alloc.o ucs/gtest-test_string.o ucs/gtest-test_sys.o ucs/gtest-test_topo.o ucs/gtest-test_sock.o ucs/gtest-test_time.o ucs/gtest-test_twheel.o ucs/gtest-test_frag_list.o ucs/gtest-test_type.o ucs/gtest-test_log.o ucs/gtest-test_iov.o ucs/gtest-test_vfs.o ucs/arch/gtest-test_x86_64.o uct/ib/gtest-test_ib.o uct/ib/gtest-test_ib_md.o uct/ib/gtest-test_cq_moderation.o uct/ib/gtest-test_ib_xfer.o uct/ib/gtest-test_ib_pkey.o uct/ib/gtest-test_ib_event.o uct/ib/gtest-test_devx.o uct/ib/gtest-ud_base.o uct/ib/gtest-test_ud.o uct/ib/gtest-test_ud_timer.o uct/ib/gtest-test_ud_pending.o uct/ib/gtest-test_ud_ds.o uct/ib/gtest-test_rc.o uct/ib/gtest-test_dc.o uct/ib/gtest-test_sockaddr.o   ../../src/ucs/libucs.la ../../src/ucs/vfs/sock/libucs_vfs_sock.la ../../src/uct/libuct.la ../../src/ucm/libucm.la ../../src/ucp/libucp.la ../../src/tools/perf/lib/libucxperf.la -fopenmp  -libverbs ../../src/uct/ib/libuct_ib.la   -lrt -lrt  -lbfd -ldl
libtool: link: /usr/bin/g++ -O3 -g -Wall -Werror -funwind-tables -Wno-missing-field-initializers -Wno-unused-parameter -Wno-unused-label -Wno-long-long -Wno-endif-labels -Wno-sign-compare -Wno-multichar -Wno-deprecated-declarations -Winvalid-pch -fno-tree-vectorize -DGTEST_UCM_HOOK_LIB_DIR=\"/home/tkloczko/rpmbuild/BUILD/ucx-1.11.1/test/gtest/ucm/test_dlopen/.libs\" -O2 -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fdata-sections -ffunction-sections -flto=auto -flto-partition=none -Wl,-dynamic-list-data -Wl,-z -Wl,relro -Wl,--as-needed -Wl,--gc-sections -Wl,-z -Wl,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -flto=auto -flto-partition=none -fuse-linker-plugin -o gtest common/gtest-gtest-all.o common/gtest-main.o common/gtest-test_gtest_cmn.o common/gtest-mem_buffer.o common/gtest-test_helpers.o common/gtest-test_obj_size.o common/gtest-test_watchdog.o common/gtest-test_perf.o common/gtest-test.o ucm/gtest-malloc_hook.o uct/gtest-test_amo.o uct/gtest-test_amo_add_xor.o uct/gtest-test_amo_and_or.o uct/gtest-test_amo_cswap.o uct/gtest-test_amo_fadd_fxor.o uct/gtest-test_amo_fand_for.o uct/gtest-test_amo_swap.o uct/gtest-test_event.o uct/gtest-test_fence.o uct/gtest-test_flush.o uct/gtest-test_many2one_am.o uct/gtest-test_md.o uct/gtest-test_mm.o uct/gtest-test_mem.o uct/gtest-test_p2p_am.o uct/gtest-test_p2p_err.o uct/gtest-test_p2p_mix.o uct/gtest-test_p2p_rma.o uct/gtest-test_pending.o uct/gtest-test_progress.o uct/gtest-test_uct_ep.o uct/gtest-test_uct_perf.o uct/v2/gtest-test_uct_query.o uct/gtest-test_zcopy_comp.o uct/gtest-uct_p2p_test.o uct/gtest-uct_test.o uct/gtest-test_stats.o ucs/gtest-test_event_set.o ucs/gtest-test_stats_filter.o uct/gtest-test_peer_failure.o uct/gtest-test_tag.o uct/tcp/gtest-test_tcp.o ucp/gtest-test_ucp_am.o ucp/gtest-test_ucp_ep.o ucp/gtest-test_ucp_stream.o ucp/gtest-test_ucp_peer_failure.o ucp/gtest-test_ucp_atomic.o ucp/gtest-test_ucp_dt.o ucp/gtest-test_ucp_tls.o ucp/gtest-test_ucp_memheap.o ucp/gtest-test_ucp_mmap.o ucp/gtest-test_ucp_mem_type.o ucp/gtest-test_ucp_perf.o ucp/gtest-test_ucp_proto.o ucp/gtest-test_ucp_rma.o ucp/gtest-test_ucp_rma_mt.o ucp/gtest-test_ucp_tag_cancel.o ucp/gtest-test_ucp_tag_match.o ucp/gtest-test_ucp_tag_offload.o ucp/gtest-test_ucp_tag_mt.o ucp/gtest-test_ucp_tag_perf.o ucp/gtest-test_ucp_tag_probe.o ucp/gtest-test_ucp_tag_xfer.o ucp/gtest-test_ucp_tag_mem_type.o ucp/gtest-test_ucp_tag.o ucp/gtest-test_ucp_context.o ucp/gtest-test_ucp_worker.o ucp/gtest-test_ucp_wireup.o ucp/gtest-test_ucp_request.o ucp/gtest-test_ucp_wakeup.o ucp/gtest-test_ucp_fence.o ucp/gtest-test_ucp_sockaddr.o ucp/gtest-ucp_test.o ucp/gtest-ucp_datatype.o ucs/gtest-test_algorithm.o ucs/gtest-test_arbiter.o ucs/gtest-test_async.o ucs/gtest-test_bitmap.o ucs/gtest-test_callbackq.o ucs/gtest-test_class.o ucs/gtest-test_config.o ucs/gtest-test_conn_match.o ucs/gtest-test_datatype.o ucs/gtest-test_bitops.o ucs/gtest-test_debug.o ucs/gtest-test_memtrack.o ucs/gtest-test_math.o ucs/gtest-test_mpmc.o ucs/gtest-test_mpool.o ucs/gtest-test_pgtable.o ucs/gtest-test_profile.o ucs/gtest-test_rcache.o ucs/gtest-test_memtype_cache.o ucs/gtest-test_stats.o ucs/gtest-test_strided_alloc.o ucs/gtest-test_string.o ucs/gtest-test_sys.o ucs/gtest-test_topo.o ucs/gtest-test_sock.o ucs/gtest-test_time.o ucs/gtest-test_twheel.o ucs/gtest-test_frag_list.o ucs/gtest-test_type.o ucs/gtest-test_log.o ucs/gtest-test_iov.o ucs/gtest-test_vfs.o ucs/arch/gtest-test_x86_64.o uct/ib/gtest-test_ib.o uct/ib/gtest-test_ib_md.o uct/ib/gtest-test_cq_moderation.o uct/ib/gtest-test_ib_xfer.o uct/ib/gtest-test_ib_pkey.o uct/ib/gtest-test_ib_event.o uct/ib/gtest-test_devx.o uct/ib/gtest-ud_base.o uct/ib/gtest-test_ud.o uct/ib/gtest-test_ud_timer.o uct/ib/gtest-test_ud_pending.o uct/ib/gtest-test_ud_ds.o uct/ib/gtest-test_rc.o uct/ib/gtest-test_dc.o uct/ib/gtest-test_sockaddr.o -fopenmp  ../../src/ucs/.libs/libucs.so ../../src/ucs/vfs/sock/.libs/libucs_vfs_sock.a ../../src/uct/.libs/libuct.so ../../src/ucm/.libs/libucm.so ../../src/ucp/.libs/libucp.so ../../src/tools/perf/lib/.libs/libucxperf.a /home/tkloczko/rpmbuild/BUILD/ucx-1.11.1/src/ucp/.libs/libucp.so ../../src/uct/ib/.libs/libuct_ib.so -libverbs -lmlx5 /home/tkloczko/rpmbuild/BUILD/ucx-1.11.1/src/uct/.libs/libuct.so /home/tkloczko/rpmbuild/BUILD/ucx-1.11.1/src/ucs/.libs/libucs.so -lnuma -lm /home/tkloczko/rpmbuild/BUILD/ucx-1.11.1/src/ucm/.libs/libucm.so -lrt -lbfd -ldl -fopenmp -Wl,-rpath -Wl,/home/tkloczko/rpmbuild/BUILD/ucx-1.11.1/src/ucs/.libs -Wl,-rpath -Wl,/home/tkloczko/rpmbuild/BUILD/ucx-1.11.1/src/uct/.libs -Wl,-rpath -Wl,/home/tkloczko/rpmbuild/BUILD/ucx-1.11.1/src/ucm/.libs -Wl,-rpath -Wl,/home/tkloczko/rpmbuild/BUILD/ucx-1.11.1/src/ucp/.libs -Wl,-rpath -Wl,/home/tkloczko/rpmbuild/BUILD/ucx-1.11.1/src/uct/ib/.libs -Wl,-rpath -Wl,/usr/lib64/ucx
ucs/test_async.cc:20: error: type 'struct base' violates the C++ One Definition Rule [-Werror=odr]
   20 | class base {
      |
ucs/test_class.cc:16: note: a type with different virtual table pointers is defined in another translation unit
   16 | typedef struct base {
      |
lto1: all warnings being treated as errors
lto-wrapper: fatal error: /usr/bin/g++ returned 1 exit status
```

Just checked and when LTO is disabled build fails but with other issues
```console
make[4]: Leaving directory '/home/tkloczko/rpmbuild/BUILD/ucx-1.11.1/test/gtest/ucm/test_dlopen'
make[3]: Leaving directory '/home/tkloczko/rpmbuild/BUILD/ucx-1.11.1/test/gtest/ucm/test_dlopen'
make[3]: Entering directory '/home/tkloczko/rpmbuild/BUILD/ucx-1.11.1/test/gtest'
/usr/bin/g++ -DHAVE_CONFIG_H -I. -I../..  -DCPU_FLAGS="" -I/home/tkloczko/rpmbuild/BUILD/ucx-1.11.1/src -I/home/tkloczko/rpmbuild/BUILD/ucx-1.11.1 -I/home/tkloczko/rpmbuild/BUILD/ucx-1.11.1/src -DTOP_SRCDIR=\"../..\" -I../../src -I../../test -I../../src -I../../test/gtest  -fopenmp     -O3 -g -Wall -Werror -funwind-tables -Wno-missing-field-initializers -Wno-unused-parameter -Wno-unused-label -Wno-long-long -Wno-endif-labels -Wno-sign-compare -Wno-multichar -Wno-deprecated-declarations -Winvalid-pch -fno-tree-vectorize -DGTEST_UCM_HOOK_LIB_DIR="\"/home/tkloczko/rpmbuild/BUILD/ucx-1.11.1/test/gtest/ucm/test_dlopen/.libs\""  -O2 -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fdata-sections -ffunction-sections -c -o ucs/gtest-test_bitops.o `test -f 'ucs/test_bitops.cc' || echo './'`ucs/test_bitops.cc
/usr/bin/g++ -DHAVE_CONFIG_H -I. -I../..  -DCPU_FLAGS="" -I/home/tkloczko/rpmbuild/BUILD/ucx-1.11.1/src -I/home/tkloczko/rpmbuild/BUILD/ucx-1.11.1 -I/home/tkloczko/rpmbuild/BUILD/ucx-1.11.1/src -DTOP_SRCDIR=\"../..\" -I../../src -I../../test -I../../src -I../../test/gtest  -fopenmp     -O3 -g -Wall -Werror -funwind-tables -Wno-missing-field-initializers -Wno-unused-parameter -Wno-unused-label -Wno-long-long -Wno-endif-labels -Wno-sign-compare -Wno-multichar -Wno-deprecated-declarations -Winvalid-pch -fno-tree-vectorize -DGTEST_UCM_HOOK_LIB_DIR="\"/home/tkloczko/rpmbuild/BUILD/ucx-1.11.1/test/gtest/ucm/test_dlopen/.libs\""  -O2 -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fdata-sections -ffunction-sections -c -o uct/ib/gtest-test_dc.o `test -f 'uct/ib/test_dc.cc' || echo './'`uct/ib/test_dc.cc
/usr/bin/g++ -DHAVE_CONFIG_H -I. -I../..  -DCPU_FLAGS="" -I/home/tkloczko/rpmbuild/BUILD/ucx-1.11.1/src -I/home/tkloczko/rpmbuild/BUILD/ucx-1.11.1 -I/home/tkloczko/rpmbuild/BUILD/ucx-1.11.1/src -DTOP_SRCDIR=\"../..\" -I../../src -I../../test -I../../src -I../../test/gtest  -fopenmp     -O3 -g -Wall -Werror -funwind-tables -Wno-missing-field-initializers -Wno-unused-parameter -Wno-unused-label -Wno-long-long -Wno-endif-labels -Wno-sign-compare -Wno-multichar -Wno-deprecated-declarations -Winvalid-pch -fno-tree-vectorize -DGTEST_UCM_HOOK_LIB_DIR="\"/home/tkloczko/rpmbuild/BUILD/ucx-1.11.1/test/gtest/ucm/test_dlopen/.libs\""  -O2 -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fdata-sections -ffunction-sections -c -o uct/ib/gtest-test_sockaddr.o `test -f 'uct/ib/test_sockaddr.cc' || echo './'`uct/ib/test_sockaddr.cc
In file included from /home/tkloczko/rpmbuild/BUILD/ucx-1.11.1/src/ucs/sys/math.h:17,
                 from /home/tkloczko/rpmbuild/BUILD/ucx-1.11.1/src/ucs/sys/string.h:12,
                 from ./common/test_helpers.h:19,
                 from ./common/test.h:19,
                 from ucs/test_bitops.cc:6:
/home/tkloczko/rpmbuild/BUILD/ucx-1.11.1/src/ucs/arch/bitops.h: In member function ‘virtual void test_bitops_is_equal_Test::test_body()’:
/home/tkloczko/rpmbuild/BUILD/ucx-1.11.1/src/ucs/arch/bitops.h:186:14: error: array subscript 20 is outside array bounds of ‘uint8_t [20]’ {aka ‘unsigned char [20]’} [-Werror=array-bounds]
  186 |     return ((*((uint8_t*)ptr1 + length) & ~UCS_MASK(8 - remainder)) ==
      |              ^~~~~~~~~~~~~~~~~~~~~~~~~~
ucs/test_bitops.cc:87:13: note: while referencing ‘buffer1’
   87 |     uint8_t buffer1[20] = {0};
      |             ^~~~~~~
In file included from /home/tkloczko/rpmbuild/BUILD/ucx-1.11.1/src/ucs/sys/math.h:17,
                 from /home/tkloczko/rpmbuild/BUILD/ucx-1.11.1/src/ucs/sys/string.h:12,
                 from ./common/test_helpers.h:19,
                 from ./common/test.h:19,
                 from ucs/test_bitops.cc:6:
/home/tkloczko/rpmbuild/BUILD/ucx-1.11.1/src/ucs/arch/bitops.h:187:14: error: array subscript 20 is outside array bounds of ‘uint8_t [20]’ {aka ‘unsigned char [20]’} [-Werror=array-bounds]
  187 |             (*((uint8_t*)ptr2 + length) & ~UCS_MASK(8 - remainder)));
      |              ^~~~~~~~~~~~~~~~~~~~~~~~~~
ucs/test_bitops.cc:88:13: note: while referencing ‘buffer2’
   88 |     uint8_t buffer2[20] = {0};
      |             ^~~~~~~
cc1plus: all warnings being treated as errors
make[3]: *** [Makefile:2698: ucs/gtest-test_bitops.o] Error 1
```

--- a/test/gtest/ucs/test_async.cc
+++ b/test/gtest/ucs/test_async.cc
@@ -17,12 +17,12 @@ extern "C" {
 #include <sys/poll.h>
 
 
-class base {
+class base_async {
 public:
-    base(ucs_async_mode_t mode) : m_mode(mode), m_count(0), m_handler_set(0) {
+    base_async(ucs_async_mode_t mode) : m_mode(mode), m_count(0), m_handler_set(0) {
     }
 
-    virtual ~base() {
+    virtual ~base_async() {
     }
 
     int count() const {
@@ -42,14 +42,14 @@ public:
     }
 
 private:
-    base(const base& other);
+    base_async(const base_async& other);
 
 protected:
     virtual void ack_event() = 0;
     virtual int event_id() = 0;
 
     static void cb(int id, ucs_event_set_types_t events, void *arg) {
-        base *self = reinterpret_cast<base*>(arg);
+        base_async *self = reinterpret_cast<base_async*>(arg);
         self->handler();
     }
 
@@ -67,9 +67,9 @@ protected:
     uint32_t               m_handler_set;
 };
 
-class base_event : public base {
+class base_event : public base_async {
 public:
-    base_event(ucs_async_mode_t mode) : base(mode) {
+    base_event(ucs_async_mode_t mode) : base_async(mode) {
         ucs_status_t status = ucs_async_pipe_create(&m_event_pipe);
         ASSERT_UCS_OK(status);
     }
@@ -84,7 +84,7 @@ public:
                                         UCS_EVENT_SET_EVREAD,
                                         cb, this, async);
         ASSERT_UCS_OK(status);
-        base::set_handler();
+        base_async::set_handler();
     }
 
     virtual int event_id() {
@@ -112,10 +112,10 @@ private:
     ucs_async_pipe_t m_event_pipe;
 };
 
-class base_timer : public base {
+class base_timer : public base_async {
 public:
     base_timer(ucs_async_mode_t mode) :
-        base(mode), m_timer_id(-1)
+        base_async(mode), m_timer_id(-1)
     {
     }
 
@@ -127,7 +127,7 @@ public:
         ucs_status_t status = ucs_async_add_timer(mode(), interval, cb,
                                                   this, async, &m_timer_id);
         ASSERT_UCS_OK(status);
-        base::set_handler();
+        base_async::set_handler();
     }
 
     virtual int event_id() {
@@ -708,7 +708,7 @@ public:
 
 protected:
     virtual void handler() {
-         base::handler();
+         base_async::handler();
          unset_handler(false);
     }
 };
@@ -732,8 +732,8 @@ public:
 
 protected:
     virtual void handler() {
-         base::handler();
-         unset_handler(m_sync);
+        base_async::handler();
+        unset_handler(m_sync);
     }
 
 private:
@@ -789,15 +789,16 @@ protected:
     }
 
     virtual void handler() {
-         base::handler();
-         if (!m_event_set) {
-             ucs_status_t status =
-                 ucs_async_set_event_handler(mode(), m_pipefd[0],
-                                             UCS_EVENT_SET_EVREAD,
-                                             dummy_cb, this, &m_async);
-             ASSERT_UCS_OK(status);
-             m_event_set = true;
-         }
+        base_async::handler();
+        if (!m_event_set) {
+            ucs_status_t status =
+                    ucs_async_set_event_handler(mode(), m_pipefd[0],
+                                                UCS_EVENT_SET_EVREAD,
+                                                dummy_cb, this,
+                                                &m_async);
+            ASSERT_UCS_OK(status);
+            m_event_set = true;
+        }
     }
 
     int m_pipefd[2];

--- a/test/gtest/ucs/test_class.cc
+++ b/test/gtest/ucs/test_class.cc
@@ -13,18 +13,18 @@ class test_class : public ucs::test {
 };
 
 
-typedef struct base {
+typedef struct {
     int            field1;
 } base_t;
 UCS_CLASS_DECLARE(base_t, int);
 
-typedef struct derived {
+typedef struct {
     base_t         super;
     int            field2;
 } derived_t;
 UCS_CLASS_DECLARE(derived_t, int, int);
 
-typedef struct derived2 {
+typedef struct {
     base_t         super;
     int            field2;
 } derived2_t;

